### PR TITLE
move _add__ and __sub__ to liboffsets, implement _Tick

### PR DESCRIPTION
**Pull request by jbrockmendel**
Last one for now.

Moving `__add__` and `__sub__` to liboffsets has been viable for a while now, just getting around to it.

Implements a dummy `_Tick` class that gets mixed in to `Tick`.  Then in `tslibs.period` we can replace `isinstance(obj, offsets.Tick)` checks with `isinstnace(obj, _Tick)`, obviating the need for the import of `tseries.offsets`.  This moves us closer to being able to import `Period` directly in `_libs.__init__`, which has been a long-time goal (see the comment there).  In conjunction with #21693 and #21690, I think that gets us the rest of the way there.

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -491,6 +510,12 @@ class BaseOffset(_BaseOffset):
         return -self + other
 
 
+class _Tick(object):
+    # dummy class to mix into tseries.Tick so that in tslibs.period we can
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx]: can u use triple quotes here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/offsets.pyx [hunk 2]
```diff
@@ -364,6 +364,25 @@ class _BaseOffset(object):
...
+        try:
+            return self.apply(other)
+        except ApplyTypeError:
+            return NotImplemented
+
+    def __sub__(self, other):
+        if isinstance(other, datetime):
+            raise TypeError('Cannot subtract datetime from offset.')
+        elif type(other) == type(self):
+            return self.__class__(self.n - other.n, normalize=self.normalize,
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx]: can you change to use ``type(self)`` here, I don't think we use the long form anywhere.

**Review by jreback — CHANGES_REQUESTED** [review]
tiny comment ping on green.

File: pandas/_libs/tslibs/offsets.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx]: its used below in `__mul__`.  change it there too while I'm at it?

Comment (jreback) [reply pandas/_libs/tslibs/offsets.pyx]: yeah might as well, its a bit more clean.

**Review by jreback — APPROVED** [review]
minor comments, ping on green.

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thank you sir!

**Pull request merged**